bin/pg_repack.c

```diff
@@ -1321,7 +1324,12 @@ repack_one_table(repack_table *table, const char *orderby)
 	{
 		const char *indexdef;
 		indexdef = getstr(indexres, j, 0);
-		elog(WARNING, "skipping invalid index: %s", indexdef);
+		if (error_on_invalid_index) {
+			elog(ERROR, "Invalid index: %s", indexdef);
+			return;
```

Comment:
Instead of calling `elog(ERROR)` and `return` it is better to call `elog(WARNING)` and `goto cleanup`.
Here is an example:
https://github.com/reorg/pg_repack/blob/a1489a1e6c15be8913ce37674a0adc74184c013f/bin/pg_repack.c#L1421